STUDIO/org.openl.rules.webstudio/webapp/pages/modules/repository/topPanel.xhtml

```diff
@@ -28,7 +28,7 @@
 
         <div class="navbar-inner">
             <h:outputLink onclick="openNewProjectDialog();return false;" styleClass="primary"
-                title="Create a new project" rendered="#{repositoryTreeState.canCreate}">
+                title="Create a new project" rendered="#{repositoryTreeState.canCreate and (repositoryTreeController.canCreateNewProject or not empty repositoryTreeController.nonFlatRepositories)}">
                 <h:outputText value="Create Project" />
             </h:outputLink>
 

```

Comment:
What about similar check for **Create Deploy Configuration** button?
We can configure repository for Deploy Configurations to use same repository as Design Repository for projects.
Deploy configurations are created in main branch.
Do we need to hide the button for this case?
If yes, should we also disable Deploy button (because it can create/modify Deploy Configuration) too?

Reply:
Probably in this case we should restrict all Deploy Configuration functionality.
What do you think?
We also can check it in Admin page and Install Wizard but WebStudio can be configured via environment variables, so I think that restricting in Admin page isn't enough.

Reply:
I will do it for deploy config as well, you are right. We need to prevent all history modification in this case

Reply:
done